utils/db_utils.py

```diff
@@ -158,4 +159,87 @@ def add_user_stats(user_data):
+            section_id = elt_data.get('section')
+            # This makes sure that only the first encounter of the section is added
+            # The near query gives closest points first, so the first one is the one we want
+            if section_id not in end_by_section:
+                end_by_section[section_id] = elt
+    
+    res_start = edb.get_analysis_timeseries_db().find(query_start)
+    start_by_section = {}
+    for elt in res_start:
+        elt_data = elt.get('data')
+        if elt_data:
+            section_id = elt_data.get('section')
+            if section_id not in start_by_section:
+                start_by_section[section_id] = elt
+    
+    vals = []
+    user_id_seen_dict = {}
+    number_user_seen = 0
+    # Now we can read every section crossing start point
+    for section_id in start_by_section:
+        matching_start = start_by_section[section_id]
+        matching_start_data = matching_start.get('data')
+        if matching_start_data is None:
+            # Something is wrong with the fetched data, shouldn't happen
+            continue
+        if 'idx' not in matching_start_data:
+            # Sometimes, idx is missing in data, not sure why
+            continue
+        matching_end = end_by_section.get(section_id)
+        if matching_end is None:
+            # This section_id didn't cross the end point
+            continue
+        matching_end_data = matching_end.get('data', {})
+        # idx allows us to check that the start section is crossed first, we do not care about trips going the other way
+        if 'idx' in matching_end_data and matching_start_data.get('idx') < matching_end_data.get('idx'):
+            user_id = str(start_by_section[section_id].get('user_id'))
+            if user_id_seen_dict.get(user_id) is None:
+                number_user_seen += 1
+                user_id_seen_dict[user_id] = True
+            vals.append({
+                'start': start_by_section[section_id], 
+                'end': end_by_section[section_id], 
+                'duration': matching_end_data.get('ts') - matching_start_data.get('ts'),
+                'section': section_id, 
+                'mode': matching_start_data.get('mode'), # Note: this is the mode given by the phone, not the computed one, we'll read it later from inferred_sections
+                'start_fmt_time': matching_start_data.get('fmt_time'),
+                'end_fmt_time': matching_end_data.get('fmt_time')
+            })
+    if perm_utils.permissions.get("segment_trip_time_min_users", 0) <= number_user_seen:
+        return pd.DataFrame.from_dict(vals)
+    return pd.DataFrame.from_dict([])
+
+# The following query can be called multiple times, so we extract index creation from the function
+analysis_timeseries_db = edb.get_analysis_timeseries_db()
+# In theory, adding an index on the cleaned_section should improve query performance
+analysis_timeseries_db.create_index([('data.cleaned_section', pymongo.ASCENDING)])
```

Comment:
index creation in mongodb is not temporary.
We should put this into `emission.core.database` just like all the other index calls.

Reply:
True, removed in commit https://github.com/e-mission/op-admin-dashboard/pull/61/commits/88c9665ff0a6f0863372e239ed8f038eadcb5770 
And added in PR https://github.com/e-mission/e-mission-server/pull/934